rollout/trafficrouting/smi/smi_test.go

```diff
@@ -0,0 +1,216 @@
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(rollout, controllerKind),
+			},
+		},
+		Spec: smiv1alpha1.TrafficSplitSpec{
+			Service: rootSvc,
+			Backends: []smiv1alpha1.TrafficSplitBackend{
+				{
+					Service: rollout.Spec.Strategy.Canary.CanaryService,
+					Weight:  canaryWeight,
+				},
+				{
+					Service: rollout.Spec.Strategy.Canary.StableService,
+					Weight:  stableWeight,
+				},
+			},
+		},
+	}
+}
+
+// Create and test actions for client
+func fakeRollout(stableSvc, canarySvc, rootSvc string, trafficSplitName string) *v1alpha1.Rollout {
+	return &v1alpha1.Rollout{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rollout",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: v1alpha1.RolloutSpec{
+			Strategy: v1alpha1.RolloutStrategy{
+				Canary: &v1alpha1.CanaryStrategy{
+					StableService: stableSvc,
+					CanaryService: canarySvc,
+					TrafficRouting: &v1alpha1.RolloutTrafficRouting{
+						SMI: &v1alpha1.SMITrafficRouting{
+							RootService:      rootSvc,
+							TrafficSplitName: trafficSplitName,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestType(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	rollout := fakeRollout("stable-service", "canary-service", "root-service", "traffic-split-name")
+	r := NewReconciler(ReconcilerConfig{
+		Rollout:        rollout,
+		Client:         client,
+		Recorder:       &record.FakeRecorder{},
+		ControllerKind: controllerKind,
+	})
+	assert.Equal(t, Type, r.Type())
+}
+
+func TestReconcileCreateNewTrafficSplit(t *testing.T) {
+	rollout := fakeRollout("stable-service", "canary-service", "root-service", "traffic-split-name")
+	client := fake.NewSimpleClientset()
+	r := NewReconciler(ReconcilerConfig{
+		Rollout:        rollout,
+		Client:         client,
+		Recorder:       &record.FakeRecorder{},
+		ControllerKind: controllerKind,
+	})
+	err := r.Reconcile(10)
+	assert.Nil(t, err)
+	actions := client.Actions()
+	assert.Len(t, actions, 2)
+	assert.Equal(t, "get", actions[0].GetVerb())
+	assert.Equal(t, "create", actions[1].GetVerb())
+
+	obj := actions[1].(core.CreateAction).GetObject()
+	ts := &smiv1alpha1.TrafficSplit{}
+	converter := runtime.NewTestUnstructuredConverter(equality.Semantic)
+	objMap, _ := converter.ToUnstructured(obj)
+	runtime.NewTestUnstructuredConverter(equality.Semantic).FromUnstructured(objMap, ts)
+
+	expectedTS := trafficSplit("traffic-split-name", rollout, 10)
+
+	assert.Equal(t, expectedTS.TypeMeta, ts.TypeMeta)
+	assert.Equal(t, expectedTS.ObjectMeta, ts.ObjectMeta)
+	assert.Equal(t, expectedTS.Spec.Service, ts.Spec.Service)
+	assert.Equal(t, expectedTS.Spec.Backends[0].Service, ts.Spec.Backends[0].Service)
+	assert.Equal(t, expectedTS.Spec.Backends[0].Weight.Value(), ts.Spec.Backends[0].Weight.Value())
+	assert.Equal(t, expectedTS.Spec.Backends[1].Service, ts.Spec.Backends[1].Service)
+	assert.Equal(t, expectedTS.Spec.Backends[1].Weight.Value(), ts.Spec.Backends[1].Weight.Value())
```

Comment:
It wasn't working because the Traffic Split spec contains Resource.Quantity pointers. (https://github.com/servicemeshinterface/smi-sdk-go/blob/master/pkg/apis/split/v1alpha1/traffic_split.go#L38)